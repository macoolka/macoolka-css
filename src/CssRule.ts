export  * from 'macoolka-css-rule'
import { StandardProperties, SvgProperties, VendorLonghandProperties } from 'csstype';
import {Rule} from 'macoolka-css-rule'
/**
 * The Provide MDN CssProperties
 * @since 0.2.0
 */
export type BaseProps = StandardProperties<string> & SvgProperties<string> & VendorLonghandProperties<string>;
/**
 * Base Rule is output that is origial css
 * All Rule will final parse to the rule
 * @desczh
 * 基本规则输出是原始的css
 * 所有的规则最后解析到这里
 * @desczh
 */
export type BaseRule = Rule<{}, {}, BaseProps, {}>