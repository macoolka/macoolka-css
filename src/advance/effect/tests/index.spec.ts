import {  ruleModule } from '../index'
import { parseProp } from '../../../CssRule';
const parse = parseProp(ruleModule);
describe('text', () => {
    it('parse mkHoverColor', () => {
        expect(parse({
            mkHoverColor: 'main',
        })).toMatchSnapshot();
        expect(parse({
            mkHoverColor: 'accent',
        })).toMatchSnapshot();
    });
    it('parse mkHoverColor', () => {
        expect(parse({
            mkHoverTextColor: 'main',
        })).toMatchSnapshot();
     /*    expect(parse({
            mkHoverTextColors: 'yellow',
        })).toMatchSnapshot(); */
    });  
    it('parse mkCollapse', () => {
        expect(parse({
            mkCollapse: true,
        })).toMatchSnapshot();
        expect(parse({
            mkCollapse: false,
        })).toMatchSnapshot();
    });  
});

