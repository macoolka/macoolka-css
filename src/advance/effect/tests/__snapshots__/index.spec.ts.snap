// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text parse mkCollapse 1`] = `
Object {
  "display": "none",
  "height": "0px",
  "transitionDelay": "0ms",
  "transitionDuration": "200ms",
  "transitionProperty": "height",
  "transitionTimingFunction": "cubic-bezier(0.4, 0, 0.2, 1)",
}
`;

exports[`text parse mkCollapse 2`] = `
Object {
  "display": "flex",
  "height": "auto",
  "transitionDelay": "0ms",
  "transitionDuration": "200ms",
  "transitionProperty": "height",
  "transitionTimingFunction": "cubic-bezier(0.4, 0, 0.2, 1)",
}
`;

exports[`text parse mkHoverColor 1`] = `
Object {
  "selector": Object {
    ":hover:not(:disabled)": Object {
      "backgroundColor": "#26a69a",
      "color": "#fff",
    },
  },
}
`;

exports[`text parse mkHoverColor 2`] = `
Object {
  "selector": Object {
    ":hover:not(:disabled)": Object {
      "backgroundColor": "#f50057",
      "color": "#fff",
    },
  },
}
`;

exports[`text parse mkHoverColor 3`] = `
Object {
  "selector": Object {
    ":hover:not(:disabled)": Object {
      "color": "#00796b",
    },
  },
}
`;
